modules/roles/store.go

```diff
@@ -0,0 +1,83 @@
+package roles
+
+import (
+	"fmt"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/state"
+	wire "github.com/tendermint/go-wire"
+)
+
+func NewPerm(role []byte) basecoin.Actor {
+	return basecoin.Actor{
+		App:     NameRole,
+		Address: role,
+	}
+}
+
+// Role - structure to hold permissioning
+type Role struct {
+	MinSigs uint32           `json:"min_sigs"`
+	Signers []basecoin.Actor `json:"signers"`
+}
```

Comment:
Okay this role structure looks good, and the associated tests seem fine. I'm wondering if there is a way to allow for more complex role interactions here though - aka - if Jae signs off a tx is valid, otherwise you would need at least two out of three between frey, bucky, rigel. Can this already be achieved using this format? Just define two different roles for the two different scenarios?